### ISSUE TITLE
IDs in <va-accordion-item> are not always unique

### ISSUE THREAD

**Issue by laflannery** [department-of-veterans-affairs/va.gov-cms#14909]
Related to previous [accordion ID ticket](https://github.com/department-of-veterans-affairs/va.gov-cms/issues/14747)

## Describe the defect
The `<va-accordion-item>` component is utilizing the entire accordion title to generate the IDs that allow deep linking into an accordion. Therefore, the ID of the accordion can at times be the same ID for 2 accordions - which when used will open both accordions and direct the user to the second, a very confusing experience for the user. [Example of this happening](https://www.va.gov/boston-vet-center/#am-i-eligible-for-vet-center-s) on a Vet Center page

### Additional Info
I have also noticed that some IDs seem to be a combination of the section they are in and the accordion title. For example, on the same Boston Vet Center page as above, the IDs for Prepare for your visit are `id="prepare-for-your-visit-accordion-item-Transportation"`. However, there is also a problem with this because when the accordion title is multiple words, the spaces are also included in the ID, which does not work (i.e. `id="prepare-for-your-visit-accordion-item-Building and parking information"`)

## To Reproduce
1. Go to [this deep link](https://www.va.gov/boston-vet-center/#am-i-eligible-for-vet-center-s) on the Boston Vet Center page
2. Notice that 2 accordions have opened instead of just one

## AC / Expected behavior

Assumptions:
- Users/Editors must be able to deep-link to a specific accordion.

Engineering notes:
- IDs appear duplicate because accordion title is truncated at 30 chars

Estimate: 5pts, timebox and return to group if it goes over

- [ ] Each accordion should have a unique ID to ensure that when used, the user is directed to the appropriate content
- [ ] Accessibility review

## Screenshots
![image](https://github.com/department-of-veterans-affairs/va.gov-cms/assets/106678594/f3527f1d-487b-4e23-a0ca-486d0f31e46c)

### Team
Please check the team(s) that will do this work.

- [ ] `CMS Team`
- [x] `Public Websites`
- [ ] `Facilities`
- [ ] `User support`
- [ ] `Accelerated Publishing`

**Comment by jilladams** [comment]
From sprint planning: noting the examples here are all Facilities pages, where titles are truncating to become IDs. This may still apply to PW pages as well (probably), so have at min added the Facilities team label vs. moving the ticket over entirely. (@mmiddaugh @xiongjaneg FYI).

@laflannery question arose in planning around why we are truncating -- is that a limitation on the a11y side somehow? To resolve this, we're considering stopping truncation altogether, for context.

**Comment by laflannery** [comment]
It is a not a requirement for accessibility, I actually thought it was a tech reason

**Comment by jilladams** [comment]
[Stack overflow says](https://stackoverflow.com/questions/584184/what-is-a-practical-maximum-length-for-html-id#:~:text=ID%20and%20NAME%20tokens%20must,notes%20%22Avoid%20fixed%20limits.%22): from a browser perspective, no technical limit.

**Comment by jilladams** [comment]
@chriskim2311 I think this is good to pull in based on Laura's note, so going to tag it into sprint 93 for you (FYI @wesrowe ), with a couple caveats: 

1. Also might get swapped for other monitoring tickets, tbd tomorrow's refinement.
2. Re: solution: I'd like to ask you to think big picture about the problem, in terms of implementation. _In a world_ where we have magillion accordions in the site, dozens of accordions on a single page could have very very similar titles, and we need accordion IDs to be unique: what's the smart, future-proofing way to ensure uniqueness? (I swore we had a ticket somewhere about adding a hash to the IDs, but I haven't been able to find it.)

**Comment by chriskim2311** [comment]
@laflannery This is ready for review, I have the PR here https://github.com/department-of-veterans-affairs/content-build/pull/1711

**Comment by laflannery** [comment]
@chriskim2311 The issue with the duplicate IDs caused by truncated titles is resolved. However, the issue mentioned in the Additional Info - regarding spaces in the ID which is breaking it is not. Should that be broken out into a separate ticket?

**Comment by chriskim2311** [comment]
@laflannery Got it sorry forgot about that one. I found the issue and I am pushing up the change.

**Comment by laflannery** [comment]
@chriskim2311 When testing the other ID ticket this morning I noticed that the issue here with the spaces in the IDs was resolved so I wanted to make sure I noted it. Thanks!

**Comment by chriskim2311** [comment]
Closing! The change is live on production with unique ids and space issue with ids is resolved

![Boston_Vet_Center___Veterans_Affairs](https://github.com/department-of-veterans-affairs/va.gov-cms/assets/42885441/91a1902c-f4c2-44bc-89f9-d5ea08f5ee50)
